File: app/src/main/kotlin/com/igorganapolsky/vibratingwatchapp/presentation/settings_screen/SetTimerPageAdapter.java
package com.igorganapolsky.vibratingwatchapp.presentation.settings_screen;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;
import com.igorganapolsky.vibratingwatchapp.presentation.settings_screen.step.SetTimerRepeatFragment;
import com.igorganapolsky.vibratingwatchapp.presentation.settings_screen.step.SetTimerTimeFragment;
import com.igorganapolsky.vibratingwatchapp.presentation.settings_screen.step.SetTimerVibrationFragment;

class SetTimerPageAdapter extends FragmentStatePagerAdapter {

    private final Fragment[] fragments = {
        new SetTimerTimeFragment(),
        new SetTimerVibrationFragment(),
        new SetTimerRepeatFragment()
    };

    SetTimerPageAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int i) {
        return fragments[i];
    }

    @Override
    public int getCount() {
        return fragments.length;
    }
}

File: app/src/main/kotlin/com/igorganapolsky/vibratingwatchapp/presentation/new_timer_creation_screen/adapter/SetTimerPageAdapter.kt
package com.igorganapolsky.vibratingwatchapp.presentation.new_timer_creation_screen.adapter

import androidx.fragment.app.Fragment
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.igorganapolsky.vibratingwatchapp.presentation.new_timer_creation_screen.view.NewTimerRepeatFragment
import com.igorganapolsky.vibratingwatchapp.presentation.new_timer_creation_screen.view.NewTimerTimeFragment
import com.igorganapolsky.vibratingwatchapp.presentation.new_timer_creation_screen.view.NewTimerVibrationFragment

/**
 * Adapter for [NewTimerTimeFragment] screen.
 */
internal class SetTimerPageAdapter(fragment: Fragment) : FragmentStateAdapter(fragment) {

    private val fragments =
        arrayOf(
            NewTimerTimeFragment(),
            NewTimerVibrationFragment(),
            NewTimerRepeatFragment()
        )

    override fun getItemCount(): Int {
        return fragments.size
    }

    override fun createFragment(position: Int): Fragment {
        return fragments[position]
    }

}
